openml/extensions/sklearn/extension.py

```diff
@@ -1268,154 +1249,50 @@ def _prediction_to_probabilities(
             user_defined_measures['wall_clock_time_millis'] = (modelfit_dur_walltime
                                                                + modelpredict_duration_walltime)
 
-        # add client-side calculated metrics. These is used on the server as
-        # consistency check, only useful for supervised tasks
-        def _calculate_local_measure(sklearn_fn, openml_name):
-            user_defined_measures[openml_name] = sklearn_fn(test_y, pred_y)
-
-        # Task type specific outputs
-        arff_datacontent = []
-
         if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
 
+            if classes is None:
+                raise TypeError("Argument classes must not be of type 'None'")
+
             try:
-                proba_y = model_copy.predict_proba(test_x)
+                proba_y = model_copy.predict_proba(X_test)
             except AttributeError:
-                proba_y = _prediction_to_probabilities(pred_y, list(model_classes))
+                proba_y = _prediction_to_probabilities(pred_y, list(classes))
+
+            if proba_y.shape[1] != len(classes):
+                # Remap the probabilities in case there was a class missing at training time
+                # By default, the classification targets are mapped to be zero-based indices to the
+                # actual classes. Therefore, the model_classes contain the correct indices to the
+                # correct probability array (the actualy array might be incorrect if there are some
+                # classes not present during train time).
+                proba_y_new = np.zeros((proba_y.shape[0], len(classes)))
+                for idx, model_class in enumerate(model_classes):
+                    proba_y_new[:, model_class] = proba_y[:, idx]
+                proba_y = proba_y_new
 
             if proba_y.shape[1] != len(task.class_labels):
-                warnings.warn(
-                    "Repeat %d Fold %d: estimator only predicted for %d/%d classes!"
-                    % (rep_no, fold_no, proba_y.shape[1], len(task.class_labels))
+                message = "Estimator only predicted for {}/{} classes!".format(
```

Comment:
in what scenario should be expect `len(classes)` to differ from `len(task.class_labels)`? (Also, what even is the difference between `classes` and `task.class_labels`?)

Reply:
There's no difference, thanks for spotting this issue, it'll make the API a bit easier.